File: app/src/main/java/com/squareoverlay/NextZoomLevelButtonView.java
package com.squareoverlay;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;

public class NextZoomLevelButtonView extends BaseButtonView {

    private static final int BUTTON_WIDTH = 280;
    private static final int BUTTON_HEIGHT = 200;
    private static final int CORNER_RADIUS = 30;

    public NextZoomLevelButtonView(Context context) {
        super(context, BUTTON_WIDTH, BUTTON_HEIGHT, CORNER_RADIUS);
        buttonTextPaint.setTextSize(50);
    }

    @Override
    protected int getGradientStartColor() {
        return Color.rgb(180, 50, 200); // Light purple
    }

    @Override
    protected int getGradientEndColor() {
        return Color.rgb(140, 20, 160); // Darker purple
    }

    @Override
    protected int getPressedColor() {
        return Color.rgb(120, 0, 120); // Darker purple when pressed
    }

    @Override
    protected void drawButtonContent(Canvas canvas) {
        float textY = buttonRect.centerY() + (buttonTextPaint.getTextSize() / 2) - 10;
        canvas.drawText("Z", buttonRect.centerX(), textY, buttonTextPaint);
    }
}

File: app/src/main/java/com/squareoverlay/NextZoomLevelButtonView.kt
package com.squareoverlay

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color

class NextZoomLevelButtonView(context: Context?) :
    BaseButtonView(context, BUTTON_WIDTH, BUTTON_HEIGHT, CORNER_RADIUS) {

    init {
        buttonTextPaint.textSize = 50f
    }

    override val gradientStartColor: Int
        get() = Color.rgb(180, 50, 200) // Light purple

    override val gradientEndColor: Int
        get() = Color.rgb(140, 20, 160) // Darker purple

    override val pressedColor: Int
        get() = Color.rgb(120, 0, 120) // Darker purple when pressed

    override fun drawButtonContent(canvas: Canvas?) {
        canvas?.let {
            val textY = buttonRect.centerY() + (buttonTextPaint.textSize / 2) - 10
            it.drawText("Z", buttonRect.centerX(), textY, buttonTextPaint)
        }
    }

    companion object {
        private const val BUTTON_WIDTH = 200
        private const val BUTTON_HEIGHT = 200
        private const val CORNER_RADIUS = 30
    }
}
